sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_utils/configurations.py

```diff
@@ -60,14 +60,22 @@ def _get_configurations(**kwargs) -> Dict[str, ConfigurationValue]:
         configurations[key] = val
     configurations[DISTRO_VERSION_ARG] = VERSION
 
-    _default_disable_logging(configurations)
-    _default_disable_metrics(configurations)
-    _default_disable_tracing(configurations)
-    _default_logger_name(configurations)
-    _default_resource(configurations)
-    _default_sampling_ratio(configurations)
-    _default_instrumentation_options(configurations)
-    _default_span_processors(configurations)
+    if "disable_logging" not in kwargs:
+        _default_disable_logging(configurations)
+    if "disable_metrics" not in kwargs:
+        _default_disable_metrics(configurations)
+    if "disable_tracing" not in kwargs:
+        _default_disable_tracing(configurations)
+    if "logger_name" not in kwargs:
+        _default_logger_name(configurations)
+    if "resource" not in kwargs:
+        _default_resource(configurations)
+    if "sampling_ratio" not in kwargs:
+        _default_sampling_ratio(configurations)
+    if "instrumentation_options" not in kwargs:
+        _default_instrumentation_options(configurations)
+    if "span_processors" not in kwargs:
+        _default_span_processors(configurations)
```

Comment:
These helper methods are designed to run always. Take the _default_resource for instance. It should run even if the resource param exists and merge the custom attributes with those from the detectors.

Is there a reason you want this change?

Reply:
can ignore this, if we make use of env variables it's working fine